File: Day237/candy135.go
package main

import (
	"fmt"
)

func candy(ratings []int) int {
	var n int = len(ratings)
	inc, dec, temp, tc := 0, 0, 0, 1
	for i:=1; i<n; i++ {
		if ratings[i-1] < ratings[i] {
			dec = 0;
			inc++
			temp = inc+1
			tc += temp
		} else if ratings[i-1] == ratings[i] {
			dec, inc, temp = 0, 0, 0
			tc++
		} else {
			dec++
			inc = 0
			tc+=dec +1
			if (temp > dec) {
				tc--
			}
		} 
	}
	return tc
}

func main() {
	arr := []int{1,0,2,5,6,4,5};
	fmt.Println(candy(arr))
}

File: Day237/Candy135.java
package Day237;

import java.util.Arrays;

public class Candy135 {

    public static int candy(int []ratings) {
        int answer = 0;
        int n = ratings.length;
        int arr[] = new int[n];
        Arrays.fill(arr, 1);
        for (int i=1; i<n; i++) {
            if (ratings[i-1]<ratings[i] && arr[i-1]>=arr[i]) {
                arr[i] = arr[i-1]+1;
            }
        }
        for (int i=n-2; i>=0; i--) {
            if (ratings[i+1]<ratings[i] && arr[i+1]>=arr[i]) {
                arr[i] = arr[i+1]+1;
            }
        }
        for (int j=0; j<n; j++) {
            answer += arr[j];
        }
        return answer;

    }
    public static void main(String[] args) {
        int nums[] = {1,0,2,5,6,4,5};
        System.out.println(candy(nums));
    }
}